### PR TITLE
algebra: adding tests for `quat_add` function

### PR DESCRIPTION
**Pull request by PNieck**
JIRA: PP-71

## Description

This PR adds tests for `quat_add` function.

## Motivation and Context

More precise tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (list targets here). `armv7a9-zynq7000-qemu`

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [x] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: algebra/tests/quat/various.c [hunk 1]
```diff
@@ -96,3 +102,58 @@ TEST_GROUP_RUNNER(group_quat_piWrite)
...
+TEST_TEAR_DOWN(group_quat_add)
+{
+}
+
+
+TEST(group_quat_add, quat_add_std)
+{
+	quat_t A = Q2;
+	quat_t B = Q3;
+	quat_t C = A; /* C is copy of A */
```
Comment (niewim19) [inline algebra/tests/quat/various.c:129]: No need for comment :) Code is sufficiently self explanatory. If you want to emphasize it even more you can write:
```c
	quat_t A = C = Q2;
	quat_t B = Q3;
```
but let`s keep it simple as it is :) Not a blocker.

**Review by niewim19 — APPROVED** [review]
LGTM

File: algebra/tests/quat/various.c (hunk 1 above)
Comment (PNieck) [reply algebra/tests/quat/various.c:129]: Ok, I have deleted comment. BTW your suggestion won't work because `C` if not defined here.

Comment (niewim19) [reply algebra/tests/quat/various.c]: Oh... yeah, my mistake. I meant:
```c
quat_t A, B, C;

A = C = Q2;
B = Q3;
```

**Review by niewim19 — APPROVED** [review]
LGTM

**Pull request merged**